src/sentry/tasks/integrations.py

```diff
@@ -19,37 +19,85 @@
 logger = logging.getLogger('sentry.tasks.integrations')
 
 
+def should_comment_sync(installation, external_issue):
+    organization = Organization.objects.get(id=external_issue.organization_id)
+    has_issue_sync = features.has('organizations:integrations-issue-sync',
+                                  organization)
+    return has_issue_sync and installation.should_sync('comment'), installation
+
+
 @instrumented_task(
-    name='sentry.tasks.integrations.post_comment',
+    name='sentry.tasks.integrations.create_comment',
     queue='integrations',
     default_retry_delay=60 * 5,
     max_retries=5
 )
 # TODO(jess): Add more retry exclusions once ApiClients have better error handling
 @retry(exclude=(ExternalIssue.DoesNotExist, Integration.DoesNotExist))
-def post_comment(external_issue_id, data, user_id, **kwargs):
-    # sync Sentry comments to an external issue
+def create_comment(external_issue_id, user_id, group_note_id, **kwargs):
     external_issue = ExternalIssue.objects.get(id=external_issue_id)
+    installation = Integration.objects.get(
+        id=external_issue.integration_id
+    ).get_installation(
+        organization_id=external_issue.organization_id,
+    )
 
-    organization = Organization.objects.get(id=external_issue.organization_id)
-    has_issue_sync = features.has('organizations:integrations-issue-sync',
-                                  organization)
-    if not has_issue_sync:
+    if not should_comment_sync(installation, external_issue):
```

Comment:
If the `should_comment_sync` function is returning a tuple, won't this condition always skip over the return?

Reply:
uh oh that was left over from something else! sorry about that